data_steward/tools/add_hpo.py

```diff
@@ -23,61 +24,60 @@
 
 LOGGER = logging.getLogger(__name__)
 
-EHR_SITE_PREFIX = 'EHR site '
-RDR = 'rdr'
-PPI_PM = 'PPI/PM'
-
-DEFAULT_DISPLAY_ORDER = """
-SELECT MAX(Display_Order) + 1 AS display_order FROM {hpo_site_id_mappings_table_id}
-"""
+DEFAULT_DISPLAY_ORDER = JINJA_ENV.from_string("""
+SELECT MAX(Display_Order) + 1 AS display_order 
+FROM `{{project_id}}.{{lookup_tables_dataset}}.{{hpo_site_id_mappings_table}}`
+""")
 
-SHIFT_HPO_SITE_DISPLAY_ORDER = """
-UPDATE {hpo_site_id_mappings_table_id}
+SHIFT_HPO_SITE_DISPLAY_ORDER = JINJA_ENV.from_string("""
+UPDATE `{{project_id}}.{{lookup_tables_dataset}}.{{hpo_site_id_mappings_table}}`
 SET Display_Order = Display_Order + 1
-WHERE Display_Order >= {display_order}
-"""
+WHERE Display_Order >= {{display_order}}
+""")
 
-ADD_HPO_SITE_ID_MAPPING = """
-SELECT '{org_id}' AS Org_ID, '{hpo_id}' AS HPO_ID, '{hpo_name}' AS Site_Name, {display_order} AS Display_Order
-"""
+ADD_HPO_SITE_ID_MAPPING = JINJA_ENV.from_string("""
+SELECT
+  '{{org_id}}' AS Org_ID, 
+  '{{hpo_id}}' AS HPO_ID, 
+  '{{hpo_name}}' AS Site_Name, 
+  {{display_order}} AS Display_Order
+""")
 
-ADD_HPO_ID_BUCKET_NAME = """
-SELECT '{hpo_id}' AS hpo_id, '{bucket_name}' AS bucket_name, '{service}' AS service
-"""
+ADD_HPO_ID_BUCKET_NAME = JINJA_ENV.from_string("""
+SELECT
+  '{{hpo_id}}' AS hpo_id, 
+  '{{bucket_name}}' AS bucket_name, 
+  '{{service}}' AS service
+""")
 
 UPDATE_SITE_MASKING_QUERY = JINJA_ENV.from_string("""
-INSERT INTO
-   `{{project_id}}.{{dataset_id}}.{{table_id}}` (hpo_id, src_id)
-     WITH random_ids AS
- (
- -- Generates the random EHR site id for the new hpo_site --
+INSERT INTO `{{project_id}}.{{dataset_id}}.{{table_id}}` (hpo_id, src_id)
+WITH available_new_src_ids AS (
+  SELECT 
+    ROW_NUMBER() OVER(ORDER BY GENERATE_UUID()) AS temp_key,
+    CONCAT('EHR site ', new_id) AS src_id
+  FROM UNNEST(GENERATE_ARRAY(100, 999)) AS new_id
+  WHERE new_id NOT IN (
+    SELECT CAST(SUBSTR(src_id, -3) AS INT64) 
+    FROM `{{project_id}}.{{dataset_id}}.{{table_id}}`
+    WHERE hpo_id != 'rdr'
+  )
+),
+hpos_not_in_site_maskings AS (
   SELECT
-    CONCAT('EHR site ', new_id) AS hpo_id,
-    ROW_NUMBER() OVER(ORDER BY GENERATE_UUID()) AS row_id
-  FROM
-    UNNEST(GENERATE_ARRAY(100, 999)) AS new_id
+    ROW_NUMBER() OVER() AS temp_key,
+    m.hpo_id
+  FROM `{{project_id}}.{{lookup_tables_dataset}}.{{hpo_site_id_mappings_table}}` AS m
+  WHERE m.HPO_ID IS NOT NULL 
+  AND m.HPO_ID != '' 
+  AND LOWER(m.HPO_ID) NOT IN (
+    SELECT LOWER(hpo_id) FROM `{{project_id}}.{{dataset_id}}.{{table_id}}`
```

Comment:
I'd suggest using `site_maskings_table` instead of `table_id` and `pipeline_tables_dataset` instead of `dataset_id`
```suggestion
    SELECT LOWER(hpo_id) FROM `{{project_id}}.{{pipeline_tables_dataset}}.{{site_maskings_table}}`
```